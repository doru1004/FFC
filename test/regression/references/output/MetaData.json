{
    "form_0": {
        "num_coefficients": 0, 
        "has_cell_integrals": 1, 
        "has_exterior_facet_integrals": 0, 
        "has_interior_facet_integrals": 0, 
        "has_point_integrals": 0, 
        "num_cell_domains": 2, 
        "num_exterior_facet_domains": 0, 
        "num_interior_facet_domains": 0, 
        "num_point_domains": 0, 
        "finite_element_0": {
            "cell_shape": 1, 
            "space_dimension": 3, 
            "value_rank": 0, 
            "evaluate_basis:_0": [1.4718852622922429e+00], 
            "evaluate_basis:_1": [2.0266705413493233e+00], 
            "evaluate_basis:_2": [-2.4985557203629258e+00], 
            "evaluate_basis_all": [1.4718852622922429e+00, 2.0266705413493233e+00, -2.4985557203629258e+00], 
            "evaluate_basis_derivatives_0_0": [1.4718852622922429e+00], 
            "evaluate_basis_derivatives_0_1": [8.5585277568607676e-02, -2.2787085025842555e+00], 
            "evaluate_basis_derivatives_0_2": [0.0, 0.0, 0.0, 0.0], 
            "evaluate_basis_derivatives_1_0": [2.0266705413493233e+00], 
            "evaluate_basis_derivatives_1_1": [-2.3642938479375970e+00, 4.4932274890774304e-01], 
            "evaluate_basis_derivatives_1_2": [0.0, 0.0, 0.0, 0.0], 
            "evaluate_basis_derivatives_2_0": [-2.4985557203629258e+00], 
            "evaluate_basis_derivatives_2_1": [2.2787085020077340e+00, 1.8293856987949413e+00], 
            "evaluate_basis_derivatives_2_2": [0.0, 0.0, 0.0, 0.0], 
            "evaluate_basis_derivatives_all_0": [1.4718852622922429e+00, 2.0266705413493233e+00, -2.4985557203629258e+00], 
            "evaluate_basis_derivatives_all_1": [8.5585277568607676e-02, -2.2787085025842555e+00, -2.3642938479375970e+00, 4.4932274890774304e-01, 2.2787085020077340e+00, 1.8293856987949413e+00], 
            "evaluate_basis_derivatives_all_2": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], 
            "evaluate_dof_0": 3.0792300000000006e-01, 
            "evaluate_dof_1": 4.3028700000000003e-01, 
            "evaluate_dof_2": 7.7282099999999998e-01, 
            "evaluate_dofs": [3.0792300000000006e-01, 4.3028700000000003e-01, 7.7282099999999998e-01], 
            "interpolate_vertex_values": [3.0792300000000006e-01, 4.3028700000000003e-01, 7.7282099999999998e-01], 
            "num_sub_elements": 0, 
        },
        "finite_element_1": {
            "cell_shape": 1, 
            "space_dimension": 3, 
            "value_rank": 0, 
            "evaluate_basis:_0": [1.4718852622922429e+00], 
            "evaluate_basis:_1": [2.0266705413493233e+00], 
            "evaluate_basis:_2": [-2.4985557203629258e+00], 
            "evaluate_basis_all": [1.4718852622922429e+00, 2.0266705413493233e+00, -2.4985557203629258e+00], 
            "evaluate_basis_derivatives_0_0": [1.4718852622922429e+00], 
            "evaluate_basis_derivatives_0_1": [8.5585277568607676e-02, -2.2787085025842555e+00], 
            "evaluate_basis_derivatives_0_2": [0.0, 0.0, 0.0, 0.0], 
            "evaluate_basis_derivatives_1_0": [2.0266705413493233e+00], 
            "evaluate_basis_derivatives_1_1": [-2.3642938479375970e+00, 4.4932274890774304e-01], 
            "evaluate_basis_derivatives_1_2": [0.0, 0.0, 0.0, 0.0], 
            "evaluate_basis_derivatives_2_0": [-2.4985557203629258e+00], 
            "evaluate_basis_derivatives_2_1": [2.2787085020077340e+00, 1.8293856987949413e+00], 
            "evaluate_basis_derivatives_2_2": [0.0, 0.0, 0.0, 0.0], 
            "evaluate_basis_derivatives_all_0": [1.4718852622922429e+00, 2.0266705413493233e+00, -2.4985557203629258e+00], 
            "evaluate_basis_derivatives_all_1": [8.5585277568607676e-02, -2.2787085025842555e+00, -2.3642938479375970e+00, 4.4932274890774304e-01, 2.2787085020077340e+00, 1.8293856987949413e+00], 
            "evaluate_basis_derivatives_all_2": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], 
            "evaluate_dof_0": 3.0792300000000006e-01, 
            "evaluate_dof_1": 4.3028700000000003e-01, 
            "evaluate_dof_2": 7.7282099999999998e-01, 
            "evaluate_dofs": [3.0792300000000006e-01, 4.3028700000000003e-01, 7.7282099999999998e-01], 
            "interpolate_vertex_values": [3.0792300000000006e-01, 4.3028700000000003e-01, 7.7282099999999998e-01], 
            "num_sub_elements": 0, 
        },
        "dofmap_0": {
            "needs_mesh_entities_0": 1, 
            "needs_mesh_entities_1": 0, 
            "needs_mesh_entities_2": 0, 
            "global_dimension": 10001, 
            "local_dimension": 3, 
            "geometric_dimension": 2, 
            "num_facet_dofs": 2, 
            "num_entity_dofs_0": 1, 
            "num_entity_dofs_1": 0, 
            "num_entity_dofs_2": 0, 
            "tabulate_dofs": [0, 0, 0], 
            "tabulate_facet_dofs_0": [1, 2], 
            "tabulate_facet_dofs_1": [0, 2], 
            "tabulate_facet_dofs_2": [0, 1], 
            "tabulate_entity_dofs_0_0": [0], 
            "tabulate_entity_dofs_0_1": [1], 
            "tabulate_entity_dofs_0_2": [2], 
            "tabulate_entity_dofs_1_0": [], 
            "tabulate_entity_dofs_1_1": [], 
            "tabulate_entity_dofs_1_2": [], 
            "tabulate_entity_dofs_2_0": [], 
            "tabulate_coordinates_0": [9.0300000000000002e-01, 3.4100000000000003e-01], 
            "tabulate_coordinates_1": [5.6100000000000005e-01, 7.6700000000000002e-01], 
            "tabulate_coordinates_2": [9.8699999999999999e-01, 7.8300000000000003e-01], 
            "num_sub_dofmaps": 0, 
        },
        "dofmap_1": {
            "needs_mesh_entities_0": 1, 
            "needs_mesh_entities_1": 0, 
            "needs_mesh_entities_2": 0, 
            "global_dimension": 10001, 
            "local_dimension": 3, 
            "geometric_dimension": 2, 
            "num_facet_dofs": 2, 
            "num_entity_dofs_0": 1, 
            "num_entity_dofs_1": 0, 
            "num_entity_dofs_2": 0, 
            "tabulate_dofs": [0, 0, 0], 
            "tabulate_facet_dofs_0": [1, 2], 
            "tabulate_facet_dofs_1": [0, 2], 
            "tabulate_facet_dofs_2": [0, 1], 
            "tabulate_entity_dofs_0_0": [0], 
            "tabulate_entity_dofs_0_1": [1], 
            "tabulate_entity_dofs_0_2": [2], 
            "tabulate_entity_dofs_1_0": [], 
            "tabulate_entity_dofs_1_1": [], 
            "tabulate_entity_dofs_1_2": [], 
            "tabulate_entity_dofs_2_0": [], 
            "tabulate_coordinates_0": [9.0300000000000002e-01, 3.4100000000000003e-01], 
            "tabulate_coordinates_1": [5.6100000000000005e-01, 7.6700000000000002e-01], 
            "tabulate_coordinates_2": [9.8699999999999999e-01, 7.8300000000000003e-01], 
            "num_sub_dofmaps": 0, 
        },
        "default_cell_integral": 0, 
        "cell_integral_0": {
            "tabulate_tensor": [1.4737277886129727e+00, -3.3607082397420962e-01, -1.1064989633924431e+00, -3.3607082397420962e-01, 1.6397204708488038e+00, -1.2724916456282744e+00, -1.1064989633924431e+00, -1.2724916456282744e+00, 2.4101486102670373e+00], 
        },
        "cell_integral_1": {
            "tabulate_tensor": [9.7209917974919235e-01, -2.2924021332143699e-01, -7.4285896642775506e-01, -2.2924021332143699e-01, 1.0827609665046967e+00, -8.5352075318326004e-01, -7.4285896642775506e-01, -8.5352075318326004e-01, 1.5963797196110150e+00], 
        },
        "default_exterior_facet_integral": 0, 
        "default_interior_facet_integral": 0, 
        "default_point_integral": 0, 
    },
}
