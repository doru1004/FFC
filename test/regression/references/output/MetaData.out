
Testing form
----------------------
0_num_coefficients = 0
1_has_cell_integrals = 1
2_has_exterior_facet_integrals = 0
3_has_interior_facet_integrals = 0
4_has_point_integrals = 0
5_num_cell_domains = 2
6_num_exterior_facet_domains = 0
7_num_interior_facet_domains = 0
8_num_point_domains = 0

Testing finite_element
----------------------
9_cell_shape = 1
10_space_dimension = 3
11_value_rank = 0
12_evaluate_basis:_0 = 1.471885262292243
13_evaluate_basis:_1 = 2.026670541349323
14_evaluate_basis:_2 = -2.498555720362926
15_evaluate_basis_all = 1.471885262292243 2.026670541349323 -2.498555720362926
16_evaluate_basis_derivatives_0_0 = 1.471885262292243
17_evaluate_basis_derivatives_0_1 = 0.08558527756860768 -2.278708502584256
18_evaluate_basis_derivatives_0_2 = 0 0 0 0
19_evaluate_basis_derivatives_1_0 = 2.026670541349323
20_evaluate_basis_derivatives_1_1 = -2.364293847937597 0.449322748907743
21_evaluate_basis_derivatives_1_2 = 0 0 0 0
22_evaluate_basis_derivatives_2_0 = -2.498555720362926
23_evaluate_basis_derivatives_2_1 = 2.278708502007734 1.829385698794941
24_evaluate_basis_derivatives_2_2 = 0 0 0 0
25_evaluate_basis_derivatives_all_0 = 1.471885262292243 2.026670541349323 -2.498555720362926
26_evaluate_basis_derivatives_all_1 = 0.08558527756860768 -2.278708502584256 -2.364293847937597 0.449322748907743 2.278708502007734 1.829385698794941
27_evaluate_basis_derivatives_all_2 = 0 0 0 0 0 0 0 0 0 0 0 0
28_evaluate_dof_0 = 0.3079230000000001
29_evaluate_dof_1 = 0.430287
30_evaluate_dof_2 = 0.772821
31_evaluate_dofs = 0.3079230000000001 0.430287 0.772821
32_interpolate_vertex_values = 0.3079230000000001 0.430287 0.772821
33_num_sub_elements = 0

Testing finite_element
----------------------
34_cell_shape = 1
35_space_dimension = 3
36_value_rank = 0
37_evaluate_basis:_0 = 1.471885262292243
38_evaluate_basis:_1 = 2.026670541349323
39_evaluate_basis:_2 = -2.498555720362926
40_evaluate_basis_all = 1.471885262292243 2.026670541349323 -2.498555720362926
41_evaluate_basis_derivatives_0_0 = 1.471885262292243
42_evaluate_basis_derivatives_0_1 = 0.08558527756860768 -2.278708502584256
43_evaluate_basis_derivatives_0_2 = 0 0 0 0
44_evaluate_basis_derivatives_1_0 = 2.026670541349323
45_evaluate_basis_derivatives_1_1 = -2.364293847937597 0.449322748907743
46_evaluate_basis_derivatives_1_2 = 0 0 0 0
47_evaluate_basis_derivatives_2_0 = -2.498555720362926
48_evaluate_basis_derivatives_2_1 = 2.278708502007734 1.829385698794941
49_evaluate_basis_derivatives_2_2 = 0 0 0 0
50_evaluate_basis_derivatives_all_0 = 1.471885262292243 2.026670541349323 -2.498555720362926
51_evaluate_basis_derivatives_all_1 = 0.08558527756860768 -2.278708502584256 -2.364293847937597 0.449322748907743 2.278708502007734 1.829385698794941
52_evaluate_basis_derivatives_all_2 = 0 0 0 0 0 0 0 0 0 0 0 0
53_evaluate_dof_0 = 0.3079230000000001
54_evaluate_dof_1 = 0.430287
55_evaluate_dof_2 = 0.772821
56_evaluate_dofs = 0.3079230000000001 0.430287 0.772821
57_interpolate_vertex_values = 0.3079230000000001 0.430287 0.772821
58_num_sub_elements = 0

Testing dofmap
----------------------
59_needs_mesh_entities_0 = 1
60_needs_mesh_entities_1 = 0
61_needs_mesh_entities_2 = 0
62_global_dimension = 10001
63_local_dimension = 3
64_geometric_dimension = 2
65_num_facet_dofs = 2
66_num_entity_dofs_0 = 1
67_num_entity_dofs_1 = 0
68_num_entity_dofs_2 = 0
69_tabulate_dofs = 0 0 0
70_tabulate_facet_dofs_0 = 1 2
71_tabulate_facet_dofs_1 = 0 2
72_tabulate_facet_dofs_2 = 0 1
73_tabulate_entity_dofs_0_0 = 0
74_tabulate_entity_dofs_0_1 = 1
75_tabulate_entity_dofs_0_2 = 2
76_tabulate_entity_dofs_1_0 =
77_tabulate_entity_dofs_1_1 =
78_tabulate_entity_dofs_1_2 =
79_tabulate_entity_dofs_2_0 =
80_tabulate_coordinates_0 = 0.903 0.341
81_tabulate_coordinates_1 = 0.5610000000000001 0.767
82_tabulate_coordinates_2 = 0.987 0.783
83_num_sub_dofmaps = 0

Testing dofmap
----------------------
84_needs_mesh_entities_0 = 1
85_needs_mesh_entities_1 = 0
86_needs_mesh_entities_2 = 0
87_global_dimension = 10001
88_local_dimension = 3
89_geometric_dimension = 2
90_num_facet_dofs = 2
91_num_entity_dofs_0 = 1
92_num_entity_dofs_1 = 0
93_num_entity_dofs_2 = 0
94_tabulate_dofs = 0 0 0
95_tabulate_facet_dofs_0 = 1 2
96_tabulate_facet_dofs_1 = 0 2
97_tabulate_facet_dofs_2 = 0 1
98_tabulate_entity_dofs_0_0 = 0
99_tabulate_entity_dofs_0_1 = 1
100_tabulate_entity_dofs_0_2 = 2
101_tabulate_entity_dofs_1_0 =
102_tabulate_entity_dofs_1_1 =
103_tabulate_entity_dofs_1_2 =
104_tabulate_entity_dofs_2_0 =
105_tabulate_coordinates_0 = 0.903 0.341
106_tabulate_coordinates_1 = 0.5610000000000001 0.767
107_tabulate_coordinates_2 = 0.987 0.783
108_num_sub_dofmaps = 0
109_default_cell_integral = 0

Testing cell_integral
----------------------
110_tabulate_tensor = 1.473727788612973 -0.3360708239742096 -1.106498963392443 -0.3360708239742096 1.639720470848804 -1.272491645628274 -1.106498963392443 -1.272491645628274 2.410148610267037

Testing cell_integral
----------------------
111_tabulate_tensor = 0.9720991797491924 -0.229240213321437 -0.7428589664277551 -0.229240213321437 1.082760966504697 -0.85352075318326 -0.7428589664277551 -0.85352075318326 1.596379719611015
112_default_exterior_facet_integral = 0
113_default_interior_facet_integral = 0
114_default_point_integral = 0
